addons/website_hr_recruitment/static/tests/tours/website_hr_recruitment.js

```diff
@@ -2,39 +2,131 @@ odoo.define('website_hr_recruitment.tour', function(require) {
     'use strict';
 
     var tour = require("web_tour.tour");
+    function applyForAJob(jobName, application) {
+        return [{
+            content: "Select Job",
+            trigger: `.oe_website_jobs h3 span:contains(${jobName})`
+        }, {
+            content: "Apply",
+            trigger: ".js_hr_recruitment a:contains('Apply')"
+        }, {
+            content: "Complete name",
+            trigger: "input[name=partner_name]",
+            run: `text ${application.name}`
+        }, {
+            content: "Complete Email",
+            trigger: "input[name=email_from]",
+            run: `text ${application.email}`
+        }, {
+            content: "Complete phone number",
+            trigger: "input[name=partner_phone]",
+            run: `text ${application.phone}`
+        }, {
+            content: "Complete Subject",
+            trigger: "textarea[name=description]",
+            run: `text ${application.subject}`
+        }, { // TODO: Upload a file ?
+            content: "Send the form",
+            trigger: ".s_website_form_send"
+        }, {
+            content: "Check the form is submited without errors",
+            trigger: ".oe_structure:has(h1:contains('Congratulations'))"
+        }];
+    }
 
     tour.register('website_hr_recruitment_tour', {
         test: true,
         url: '/jobs',
+    }, [].concat(
+    applyForAJob('Guru',
+        {
+            name: 'John Smith',
+            email: 'john@smith.com',
+            phone: '118.218',
+            subject: '### [GURU] HR RECRUITMENT TEST DATA ###',
+    }), {
+            content: "Go back to the jobs page",
+            trigger: "body",
+            run: function () {
+                window.location.href = '/jobs';
+            }
+    },
+    applyForAJob('Internship',
+        {
+            name: 'Jack Doe',
+            email: 'jack@doe.com',
+            phone: '118.712',
+            subject: '### HR [INTERN] RECRUITMENT TEST DATA ###',
+        }),
+    ));
```

Comment:
Hard to read in review but your use of concat seems off or maybe I am confused by the indentation. Aren't you giving an object as second argument? How can that work?